cmd/tether/tether_test.go

```diff
@@ -109,9 +109,15 @@ func (t *Mocker) SessionLog(session *tether.SessionConfig) (dio.DynamicMultiWrit
 	return dio.MultiWriter(&t.SessionLogBuffer, os.Stdout), nil
 }
 
-func (t *Mocker) HandleSessionExit(config *tether.ExecutorConfig, session *tether.SessionConfig) bool {
+func (t *Mocker) HandleSessionExit(config *tether.ExecutorConfig, session *tether.SessionConfig) func() {
 	// check for executor behaviour
-	return session.ID == config.ID
+	if session.ID == config.ID {
```

Comment:
It may be cleaner to do this check inside the lambda so this `HandleSessionExit` never returns nil.